### ISSUE TITLE
QUiLoader segfaults with static build

### ISSUE THREAD

**Issue by gen2brain** [therecipe/qt#55]
Calculator example segfaults when compiled with static build of Qt, line_edits example works correctly but calculator fails:
```
fatal error: unexpected signal during runtime execution
[signal SIGSEGV: segmentation violation code=0x1 addr=0x1 pc=0x7f647e867d3a]

runtime stack:
runtime.throw(0x262c400, 0x2a)
    /home/milann/go/src/runtime/panic.go:566 +0x95
runtime.sigpanic()
    /home/milann/go/src/runtime/sigpanic_unix.go:12 +0x2cc

goroutine 1 [syscall, locked to thread]:
runtime.cgocall(0x15644b0, 0xc420055e08, 0x0)
    /home/milann/go/src/runtime/cgocall.go:131 +0x110 fp=0xc420055dd8 sp=0xc420055d98
github.com/therecipe/qt/uitools._Cfunc_QUiLoader_NewQUiLoader(0x0, 0x0)
    github.com/therecipe/qt/uitools/_obj/_cgo_gotypes.go:618 +0x4a fp=0xc420055e08 sp=0xc420055dd8
github.com/therecipe/qt/uitools.NewQUiLoader(0x0, 0x0, 0x0)
    /home/milann/golang/src/github.com/therecipe/qt/uitools/uitools.go:60 +0xae fp=0xc420055e68 sp=0xc420055e08
main.NewCalculatorForm(0x1)
    /home/milann/golang/src/github.com/therecipe/qt/internal/examples/uitools/calculator/calculator.go:23 +0x6e fp=0xc420055f10 sp=0xc420055e68
main.main()
    /home/milann/golang/src/github.com/therecipe/qt/internal/examples/uitools/calculator/calculator.go:15 +0x56 fp=0xc420055f48 sp=0xc420055f10
runtime.main()
    /home/milann/go/src/runtime/proc.go:183 +0x1f4 fp=0xc420055fa0 sp=0xc420055f48
runtime.goexit()
    /home/milann/go/src/runtime/asm_amd64.s:2086 +0x1 fp=0xc420055fa8 sp=0xc420055fa0

goroutine 17 [syscall, locked to thread]:
runtime.goexit()
    /home/milann/go/src/runtime/asm_amd64.s:2086 +0x1
```

I used this bash script for building https://gist.github.com/gen2brain/54e295da2a5c9d4a5564260fbc930256 , and I also added plugins_linux.cpp file, to include plugins in static build https://gist.github.com/gen2brain/7025c563500d34f1d23d32ee4ae84b59 .

Both binaries are here, calculator is with debug symbols:
http://81.4.106.254/calculator.gz
http://81.4.106.254/line_edits.gz

**Comment by therecipe** [comment]
Hey

I will look into this.

**Comment by gen2brain** [comment]
Thanks. I also managed to build static binaries for windows with mingw toolchain, in linux. Again line_edits example works but calculator spits same error. I tested it with wine though. 

Build script https://gist.github.com/gen2brain/ebce076e11fcbe3e6aaa20513b778f24 and plugins_windows.cpp https://gist.github.com/gen2brain/0a23b5fa2dbe7c3ba47e63aeef859eff . 

Windows binaries are fully static:
http://81.4.106.254/calculator.exe.zip
http://81.4.106.254/line_edits.exe.zip

**Comment by therecipe** [comment]
I'm currently setting up a arch linux box and will try to reproduce the error.
But if I would have to guess, then I would say that there is a problem with the link order in your CGO_LDFLAGS.
You probably need to place `-lQt5UiTools` in the very beginning, even before `-lQt5Core`.

**Comment by gen2brain** [comment]
I just tried to place -lQt5UiTools on the beginning, nothing changed. Linux binary is semi static but Windows binary is static, compiler would complain if anything is missing I think (and it does complain about many things at start) .
libQt5UiTools comes as static library even when official binary is used (don't know how it is on Windows), I don't think I saw any binary linked to that lib.

**Comment by therecipe** [comment]
Got it working on linux :)

I had to also link against the icu libs, but you probably configured Qt with `-no-icu`?
And I used the "minimal" version, created with "qtdeploy build desktop calculator minimal", so I had to add the "minimal" tag to the bash file.

Beside these changes to make-static.bash, I only had to make sure the calculator folder contains the qrc files (qrc.go, qrc.cpp and qrc.qrc) and the plugins_linux.cpp.

And I also had to change the link order in `qt/uitools/cgo_linux_amd64.go` from
`-lQt5UiTools -lQt5Core -lQt5Gui -lQt5Widgets -lpthread` to
`-lQt5UiTools -lQt5Widgets -lQt5Gui -lQt5Core -lpthread`

I probably have messed up the link order for static builds in general, but I will fix it.
